src/olympia/devhub/views.py

```diff
@@ -1166,11 +1167,31 @@ def version_edit(request, addon_id, addon, version_id):
                         'max' in form.changed_data):
                     _log_max_version_change(addon, version, form.instance)
 
-        fields = ('source', 'approvalnotes')
-        has_changed = [field in version_form.changed_data for field in fields]
-        if version.has_info_request and any(has_changed):
-            version.update(has_info_request=False)
-            version.save()
+        if 'approvalnotes' in version_form.changed_data:
+            if version.has_info_request:
+                version.update(has_info_request=False)
+                if waffle.switch_is_active('activity-email'):
+                    log_and_notify(amo.LOG.APPROVAL_NOTES_CHANGED,
+                                   _('Approval notes were updated.'),
+                                   request.user,
+                                   version)
+                else:
+                    amo.log(amo.LOG.APPROVAL_NOTES_CHANGED,
```

Comment:
Add a check we logged the note in `test_update_approvalnotes_should_drop_info_request_flag`